amads/io/scoreread.py

```diff
@@ -0,0 +1,119 @@
+"""scoreread.py -- file input"""
+
+__author__ = "Roger B. Dannenberg"
+
+import pathlib
+
+from amads.core.basics import Score
+
+preferred_reader = "music21"  # default readers
+
+
+def _check_alternate_subsystem() -> bool:
+    """Check if the alternate reader is available. If so, install it.
+
+    Returns:
+    bool: True if a reader is available, False otherwise.
+    """
+    global preferred_reader
+    original_reader = preferred_reader
+    alternate_reader = "partitura" if preferred_reader == "music21" else "music21"
+    print(f"In scoreread: {original_reader} is preferred reader,", "but not available.")
+    print(f"Attepting to use {alternate_reader} instead.")
+    preferred_reader = alternate_reader
+    if _check_for_subsystem(_try_alternate=False):
+        print(f"{preferred_reader} is available, using it instead.")
+        return True
+    else:
+        print(f"{preferred_reader} is not available either.")
+        preferred_reader = original_reader
+        return False
+
+
+def _check_for_subsystem(_try_alternate: bool = True) -> bool:
+    """Check if the preferred reader is available. If so, install it.
+    If not, try the alternate reader if _try_alternate is True.
+
+    Returns:
+    midi_import, xml_import: functions for importing MIDI and XML files
+    """
+    if preferred_reader == "music21":
+        try:
+            print("In scoreread: importing for music21")
+            from amads.io.m21_midi_import import music21_midi_import
+            from amads.io.m21_xml_import import music21_xml_import
+
+            return music21_midi_import, music21_xml_import
+        except ImportError:
```

Comment:
This is not ideal, I think, because it doesn't check that the `ImportError` corresponds specifically to `music21`; it could be another package that is missing;